README.md

```diff
@@ -112,3 +121,138 @@ make
+
+The pin layout for ports A and B is as follows:
+
+| Index | Function |
+|-------|----------|
+| 0     | SCL      |
+| 1     | SDA      |
+| 2     | SCK      |
+| 3     | MISO     |
+| 4     | MOSI     |
+| 5     | TX       |
+| 6     | RX       |
+| 7     | I/O      |
+
+All pins (0-7) can be used as digital pins. All pins on port B and pin 7 on port A can be used as analog pins.
+
+If you're newer to hardware and these functions look like alphabet soup to you, take a look at our [communication protocols documentation](https://tessel.io/docs/communicationProtocols) to get an idea of how these pins should be used.
+
+### Digital pins
+
+A digital pin (any pin other than 3.3V and GND on Tessel 2) is either high (on/3.3V) or low (off/0V). If unset, Tessel's pins are pulled high by default.
+
+Here is an example usage of a digital pin on Tessel:
+
+```js
+var tessel = require('tessel'); // import tessel
+var myPort = tessel.port['A']; // select the GPIO port
+var myPin = myPort.pin[2]; // select pin 2
+myPin.output(1);  // turn pin high (on)
+console.log(myPin.read()); // print the pin value to the console
+myPin.output(0);  // turn pin low (off)
+```
+
+### Analog pins
+
+An analog pin is a pin that can vary in the range between 3.3V and 0V. Pin 7 on port A and all pins on port B can read and write analog values (though pins 0 and 1 are not recommended for this purpose).
+
+Here is an example usage of an analog pin on Tessel:
+```js
+var tessel = require('tessel'); // import tessel
+var myPort = tessel.port['A']; // select the GPIO port
+var myPin = myPort.pin[7]; // select pin 2
+myPin.output(.6);  // turn pin to 60% of high
+console.log(myPin.read()); // print the pin value to the console
+```
+
+### PWM pins
+
+PWM pins are not yet implemented. See [#21](https://github.com/tessel/t2-firmware/issues/21).
+
+### SPI
+
+A SPI channel uses the SCK, MISO, and MOSI pins (2, 3, and 4 on Tessel 2). If you are unfamiliar with the SPI protocol, please see the [communication protocols tutorial](https://tessel.io/docs/communicationProtocols#spi).
+
+Here is an example using Tessel's SPI protocol:
+
+```js
+var port = tessel.port['A'];
+var spi = new port.SPI({
+  clockSpeed: 4*1000*1000, // 4MHz
+  cpol: 1, // polarity
+  cpha: 0, // clock phase
+});
+
+spi.transfer(new Buffer([0xde, 0xad, 0xbe, 0xef]), function (err, rx) {
+  console.log('buffer returned by SPI slave:', rx);
+});
+```
+
+### I2C
+
+An I2C channel uses the SCL and SDA pins (0 and 1 on Tessel 2). If you are unfamiliar with the I2C protocol, please see the [communication protocols tutorial](https://tessel.io/docs/communicationProtocols#i2c).
+
+Here is an example using Tessel's I2C protocol:
+
+```js
+var port = tessel.port['A'];
+var slaveAddress = 0xDE;
+var i2c = new port.I2C(slaveAddress)
+i2c.transfer(new Buffer([0xde, 0xad, 0xbe, 0xef]), function (err, rx) {
+  console.log('buffer returned by I2C slave ('+slaveAddress.toString(16)+'):', rx);
+})
+```
```

Comment:
This section should come before SPI, to keep the pin numbers in order. 

- I2C (pins 0, 1)
- SPI (pins 2, 3, 4)
- UART (pins 5, 6)